src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteHandlers/DisallowNonParsableComplexTypesOnParameters.cs

```diff
@@ -86,10 +88,36 @@ public partial class RouteHandlerAnalyzer : DiagnosticAnalyzer
             }
         }
 
-        static bool HasAttributeImplementingFromMetadataInterfaceType(IParameterSymbol parameter, WellKnownType fromMetadataInterfaceType, WellKnownTypes wellKnownTypes)
+        static string ResolveRouteParameterName(IParameterSymbol parameterSymbol, WellKnownTypes wellKnownTypes)
+        {
+            if (!TryGetFromMetadataInterfaceTypeAttributeData(parameterSymbol, WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromRouteMetadata, wellKnownTypes, out var attributeData))
+            {
+                return parameterSymbol.Name; // No route metadata attribute!
+            }
+
+            foreach (var namedArgument in attributeData.NamedArguments)
+            {
+                if (namedArgument.Key == "Name")
+                {
+                    var routeParameterNameConstant = namedArgument.Value;
+                    var routeParameterName = (string)routeParameterNameConstant.Value!;
+                    return routeParameterName; // Have attribute & name is specified.
+                }
+            }
+
+            return parameterSymbol.Name; // We have the attribute, but name isn't specified!
+        }
+
+        static bool TryGetFromMetadataInterfaceTypeAttributeData(IParameterSymbol parameterSymbol, WellKnownType fromMetadataInterfaceType, WellKnownTypes wellknowTypes, out AttributeData attributeData)
+        {
+            var fromMetadataInterfaceTypeSymbol = wellknowTypes.Get(fromMetadataInterfaceType);
+            attributeData = parameterSymbol.GetAttributes().SingleOrDefault(ad => ad.AttributeClass.Implements(fromMetadataInterfaceTypeSymbol))!;
```

Comment:
```suggestion
            attributeData = parameterSymbol.GetAttributes().FirstOrDefault(ad => ad.AttributeClass.Implements(fromMetadataInterfaceTypeSymbol))!;
```
It's possible that two attributes implement `IFromRouteMetadata`. Be defensive and use the first one. A unit test with multiple attribute matches would be good as well.

Reply:
My PR resolves this.